tests/utils.py

```diff
@@ -59,7 +59,7 @@ def check_generated_data(samples, direct_sampling=0, use_numpy=True):
         a = np.fromiter((complex(i, q) for i, q in iter_test_samples(test_len)), dtype='complex')
         a /= (255/2)
         a -= (1 + 1j)
-        assert np.array_equal(samples, a)
+        assert np.allclose(samples, a)
```

Comment:
I do not like that we have to change test code for this to work.
Even though the values from what I've seen are only off by around 1.4e-16, it's still doesn't feel right to me.